### PR TITLE
feat(cmf/connect): dispatchActionCreator can be called without context

### PR DESCRIPTION
**Pull request by jmfrancois**
it will pass the full context (registry, router, store)

**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our
  [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)

if you want to dispatch an action creator in your connected componetn you must push a context with minimal context to have the registry (to find the actioncretaor)

And we already have a full context in CMFContainer so ...

**What is the new behavior?**

the context is optional and you should not passed it except if you know what you are doing (for example you want to add a new context

**Does this PR introduce a breaking change?**

- [x] No

### CODE REVIEW

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://294.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://294.talend.surge.sh)

**Comment by jsomsanith** [comment]
Can yo uexplain why do we let the user pass another context instead of always passing the `this.context` ?

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://294.talend.surge.sh)

File: packages/cmf/src/cmfConnect.js [hunk 1]
```diff
@@ -176,5 +185,7 @@ export default function cmfConnect({
 			mergeProps,
 			{ ...rest },
 		)(hoistStatics(CMFContainer, WrappedComponent));
+		Connected.CMFContainer = CMFContainer;
```
Comment (jsomsanith) [inline packages/cmf/src/cmfConnect.js]: Why is this needed ?

File: packages/cmf/src/cmfConnect.js [hunk 2]
```diff
@@ -121,43 +121,52 @@ export default function cmfConnect({
...
-						this.context
 					);
 				}
 				if (!keepComponentState) {
 					this.props.deleteState();
 				}
 			}
 
+			dispatchActionCreator(actionCreatorId, event, data, context) {
+				const myContext = context || this.context;
```
Comment (jsomsanith) [inline packages/cmf/src/cmfConnect.js]: As discussed together, this should merge the cmf context with the user context to not break cmf internal process which need the registry in context.

File: packages/cmf/src/cmfConnect.js (hunk 1 above)
Comment (jmfrancois) [inline packages/cmf/src/cmfConnect.js]: To being able to tests this class ... it's a bit like a singleton you can't access because there is no link from outside.

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://294.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://294.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://294.talend.surge.sh)
